From fc35451acb9e42075d082bf817a10878054b7186 Mon Sep 17 00:00:00 2001
From: mayi <3363523343@qq.com>
Date: Wed, 21 Oct 2026 18:34:15 +0800
Subject: [PATCH] fix(osdrv/atbm6461): disable automatic unload for always-on
 BF01

Make the BF01 vendor supplicant default to rmmod_auto=0 while preserving the original default for other builds. No GPIO power or low-power wake behavior is added.
---
 extdrv/wireless/realtek/atbm6461/tools/Makefile | 2 +-
 extdrv/wireless/realtek/atbm6461/tools/tools.c  | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/extdrv/wireless/realtek/atbm6461/tools/Makefile b/extdrv/wireless/realtek/atbm6461/tools/Makefile
index cb041199f3e2e6120b3d2a0e1d996354096861b1..51fcd5bd85ad4e394091f065467db3b97ab1b5e5 100644
--- a/extdrv/wireless/realtek/atbm6461/tools/Makefile
+++ b/extdrv/wireless/realtek/atbm6461/tools/Makefile
@@ -1,7 +1,7 @@
 include cv181x.config
 
 WIFI_INSTALL_DIR ?= driver_install
-TOOL_CFLAGS := -Itools -Ihal_apollo -DARES_B=10 -DHERA=11 -DRHEA=12 -DEROS=13 -DPROJ_TYPE=RHEA
+TOOL_CFLAGS := -Itools -Ihal_apollo -DARES_B=10 -DHERA=11 -DRHEA=12 -DEROS=13 -DPROJ_TYPE=RHEA -DATBM_BF01_ALWAYS_ON
 
 ifeq ($(CONFIG_ATBM_SDIO_ATCMD),y)
 TOOL_CFLAGS += -DCONFIG_ATBM_SDIO_ATCMD
diff --git a/extdrv/wireless/realtek/atbm6461/tools/tools.c b/extdrv/wireless/realtek/atbm6461/tools/tools.c
index 4cd49405aac4da941607a2787571123034e3e6ef..1e9bf3e5318062221d407dee971f5a83551d9fac 100644
--- a/extdrv/wireless/realtek/atbm6461/tools/tools.c
+++ b/extdrv/wireless/realtek/atbm6461/tools/tools.c
@@ -112,7 +112,11 @@ static char ap_key_set = 0;
 static char ap_key_id_set = 0;
 static char ap_key_mgmt_set = 0;
 static char driver_can_rmmod = 0;
+#ifdef ATBM_BF01_ALWAYS_ON
+static char rmmod_auto = 0;
+#else
 static char rmmod_auto = 1;
+#endif
 static char driver_switch = 0;
 static char ipc_server_set = 0;
 static char ipc_port_set = 0;
